server/app/services/barman-service.js

```diff
@@ -197,10 +267,90 @@ async function deleteBarmanById(barmanId) {
     return barman;
 }
 
+/**
+ * Get the services of a barman
+ *
+ * @param barmanId {number} barman id
+ * @param startDate {Date} starting date for services
+ * @param endDate {Date} ending date for services
+ * @returns {Promise<Array>} Barmen's services
+ */
+async function getBarmanServices(barmanId, startDate, endDate) {
+
+    logger.verbose('Barman service: retreive services of the barman ', barmanId );
+
+    const barman = await Barman.findById(barmanId);
+
+    if (!barman) throw createUserError('UnknownBarman', 'This Barman does not exist');
+    const Op = sequelize.Op;
+    const services = await barman.getServices({
+        where: {
+            startAt : {
+                [Op.gte]: startDate,
+            }, endAt : {
+                [Op.lte]: endDate
+            }
+        }
+    });
+
+    return services;
+}
+
+/**
+* Create a service for a barman
+*
+* @param barmanID {number} barman id
+* @param serviceId {number<Array>} service ids
+* @returns {Promise<Errors.ValidationError>} an error or nothing
+*/
+async function createServiceBarman(barmanId, servicesId) {
+
+    logger.verbose('Barman service: create a Service for the barman ', barmanId, servicesId);
+
+    const barman = await Barman.findById(barmanId);
+
+    if (!barman) throw createUserError('UnknownBarman', 'This Barman does not exist');
+
+    for (const id of servicesId) {
+        const service = await Service.findById(id);
+        if (!service) throw createUserError('UnknownService', 'This service does not exist');
+        await barman.addService(servicesId);
+    }
+}
+
+/**
+* Delete a Service of a barman
+*
+* @param barmanId {number} barman id
+* @param serviceId {number} service id
+*
+* @returns {Promise<Errors.ValidationError>} an error or nothing
+*/
+async function deleteServiceBarman(barmanId, servicesId) {
+
+    const barman = await Barman.findById(barmanId);
+
+    if (!barman) throw createUserError('UnknownBarman', 'This Barman does not exist');
+
+    for (const id of servicesId) {
+        const service = await Service.findById(id);
+        if (!service) throw createUserError('UnknownService', 'This service does not exist');
+        await service.destroy({
```

Comment:
Why delete the service ? You only want to delete the association between Sevice & Barman.
Also, you shouldn't use a `for` for this, as it will make a lot of SQL requests (at least 2/item).

Instead, look at what we wrote for `roles` and `kommissions`:
- Check if the barman exists
- Use the `barman.removeService([idService1, idService2, ...])` function.

Reply:
should I use a transaction ?